Update orchestration doc (#2392)
elastic/cloud-on-k8s: docs/orchestration.asciidoc

* Elasticsearch `Pods` may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local `PersistentVolumes`. If the Kubernetes node bound to a local `PersistentVolume` does not have enough capacity to host an upgraded `Pod` which was temporarily removed, that `Pod` will stay Pending.

* Rolling upgrades can only make progress if the Elasticsearch cluster health is green. ECK might also make progress if the cluster health is yellow under the following conditions:
** A cluster version upgrade is in progress and some `Pods` are not up to date
** There are no initializing or relocating shards

> If the above conditions are met then ECK can delete a `Pod` for upgrade even if the cluster health is yellow as long as the `Pod` is not holding the last available replica of a shard.

Reviewer: ```suggestion
If the above conditions are met, then ECK can delete a `Pod` for upgrade even if the cluster health is yellow as long as the `Pod` is not holding the last available replica of a shard.
```

> Advanced users can force an upgrade by manually deleting `Pods`. The deleted `Pods` are automatically recreated at the latest revision. There are two exceptions to this rule:

Reviewer 2: "Advanced users can force an upgrade by manually deleting `Pods`. The deleted `Pods` are automatically recreated at the latest revision."

should be moved elsewhere, correct? I think it does not follow with "There are two exceptions to this rule:" because deleting a pod manually will always recreate it with the new config

> Advanced users may force an upgrade by manually deleting `Pods` themselves. The deleted `Pods` will be automatically recreated at the latest revision. There are two exceptions to this rule:

Reviewer: ```suggestion
Advanced users can force an upgrade by manually deleting `Pods`. The deleted `Pods` are automatically recreated at the latest revision. There are two exceptions to this rule:
```

* Elasticsearch `Pods` may stay `Pending` during a rolling upgrade if the Kubernetes scheduler cannot re-schedule them back. This is especially important when using local `PersistentVolumes`. If the Kubernetes node bound to a local `PersistentVolume` does not have enough capacity to host an upgraded `Pod` which was temporarily removed, that `Pod` will stay Pending.

> * Rolling upgrades can only make progress if the Elasticsearch cluster health is green. ECK might also make progress if the cluster health is yellow under the following conditions:

Reviewer 2: "ECK might" should be "ECK will"?

Also I wonder about the "only" in this sentence:
"Rolling upgrades can only make progress if the Elasticsearch cluster health is green."
since we immediately give other situations where it will make progress if it is not green. Maybe, "In most situations, rolling upgrades will only make progress if the Elasticsearch cluster health is green"?

** A cluster version upgrade is in progress and some `Pods` are not up to date
** There are no initializing or relocating shards

If the above conditions are met, then ECK can delete a `Pod` for upgrade even if the cluster health is yellow as long as the `Pod` is not holding the last available replica of a shard.

> There are two situations where the health of the cluster is deliberately ignored:

Reviewer: ```suggestion
The health of the cluster is deliberately ignored in the following cases:
```
